### ISSUE TITLE
System tables access requires extraneous permissions

### ISSUE THREAD

**Issue by jon-wei** [apache/incubator-druid#7563]
### Affected Version

0.13.0, 0.14.0

### Description

SQL queries on `sys.*` tables currently require the following permission:

```
  {
    "resource": {
      "name": "sys",
      "type": "DATASOURCE"
    },
    "action": "READ"
  }
```

In one sense, this is incorrect since "sys" is a schema and not a datasource-like table, so the permission needed should really be on something like "sys.segments". This occurs because the following block in DruidPlanner only checks the first part of the table's qualified name:

```
            if (node instanceof Bindables.BindableTableScan) {
              Bindables.BindableTableScan bts = (Bindables.BindableTableScan) node;
              RelOptTable table = bts.getTable();
              String tableName = table.getQualifiedName().get(0);
              datasourceNames.add(tableName);
            }
```

In another sense, treating these system tables as a "datasource" is not consistent with the equivalent non-SQL APIs. 

Retrieving information about a datasource's segments via MetadataResource APIs only requires datasource-read permissions, and not a higher level "view segment information" permission, which is what a permission on "sys.segments' would represent. 

The non-SQL APIs only require additional STATE-read permissions for APIs involving server information (http://druid.io/docs/latest/development/extensions-core/druid-basic-security.html#defining-permissions has more info).

The permissions needed for system table access could be adjusted to not require additional permissions beyond datasource-specific permissions for non-server related info (segments, tasks) and to require STATE-read permissions for server-related tables (servers, server_segments) to be consistent with the non-SQL APIs.

It's worth mentioning here that there is some inconsistency in the non-SQL task APIs themselves. The task APIs in OverlordResource only require datasource permissions, but the running tasks do contain information about where they exist (server-related info). On the other hand, retrieving task info from the middle managers via WorkerResource requires STATE but not datasource permissions.

**Comment by gianm** [comment]
> The permissions needed for system table access could be adjusted to not require additional permissions beyond datasource-specific permissions for non-server related info (segments, tasks) and to require STATE-read permissions for server-related tables (servers, server_segments) to be consistent with the non-SQL APIs.

IMO it makes the most sense to align the permissions for SQL system tables with the permissions for similar metadata APIs as closely as possible. So, doing the above.

> It's worth mentioning here that there is some inconsistency in the non-SQL task APIs themselves. The task APIs in OverlordResource only require datasource permissions, but the running tasks do contain information about where they exist (server-related info). On the other hand, retrieving task info from the middle managers via WorkerResource requires STATE but not datasource permissions.

One major issue here is that nothing really defines what "STATE" and "CONFIG" are supposed to mean (other than access to an enumerated list of endpoints). My general feeling is that if you have "STATE" permissions it's reasonable to be able to see any sort of metadata (including datasources you don't have specifically have read access grants for). If you couldn't, then you'd presumably get partial metadata, which would be a misleading view of cluster state. But that isn't written down anywhere that I can see, so it's a missing aspect to the permissions model and the documentation.

In the specific case you brought up, the only change I might make (given the above reasonable) is expanding the permissions such that you can see all tasks, for any datasource, if you have "STATE" permissions.

**Comment by gianm** [comment]
Fixed by #7579.
